frontend/packages/operator-lifecycle-manager/src/components/operand.tsx

```diff
@@ -167,8 +167,57 @@ export const OperandTableHeader = () => {
   ];
 };
 
+export enum OperatorStatusType {
+  phase = 'phase',
+  status = 'status',
+  state = 'state',
+  conditions = 'conditions',
+}
+
+export const OperatorStatusTypeText = {
+  [OperatorStatusType.phase]: 'Phase:',
+  [OperatorStatusType.status]: 'Status:',
+  [OperatorStatusType.state]: 'State:',
+  [OperatorStatusType.conditions]: 'Condition:',
+};
+
+export const OperandStatusIconAndText: React.FunctionComponent<OperandStatusIconAndTextProps> = ({
+  statusObject,
+}) => {
+  let iconAndText = <div className="text-muted">Unknown</div>;
+  if (_.isEmpty(statusObject)) {
+    return iconAndText;
+  }
+  _.forEach(OperatorStatusType, (key: string) => {
+    if (_.has(statusObject, key)) {
+      const status = statusObject[key];
+      if (!_.isEmpty(status)) {
+        let statusIcon;
+        if (key !== OperatorStatusType.conditions) {
+          if (status === 'Running') {
```

Comment:
I'd use a ternary here to avoid too much nesting

```ts
statusIcon = status === 'Running' ? <SuccessStatus title={status} /> : <Status status={status} />;
```